extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/devservices/DevServicesJaegerProcessor.java

```diff
@@ -0,0 +1,248 @@
+            return null;
+        }
+
+        if (!dockerStatusBuildItem.isDockerAvailable()) {
+            log.warn("Please configure '" + OTEL_CONFIG_ENDPOINT + "' or get a working docker instance");
+            return null;
+        }
+
+        DockerImageName dockerImageName = DockerImageName
+                .parse(capturedDevServicesConfiguration.imageName().get());
+
+        final Supplier<RunningDevService> defaultJaegerSupplier = () -> {
+            QuarkusJaegerContainer jaegerContainer = new QuarkusJaegerContainer(dockerImageName,
+                    capturedDevServicesConfiguration.consolePort(),
+                    capturedDevServicesConfiguration.exporterPort(),
+                    capturedDevServicesConfiguration.serviceName(), useSharedNetwork);
+            timeout.ifPresent(jaegerContainer::withStartupTimeout);
+            jaegerContainer.start();
+
+            String jaegerHost = JAEGER_SCHEME + jaegerContainer.getHost() + ":" + jaegerContainer.getPort();
+
+            return new RunningDevService(Feature.OPENTELEMETRY_JAEGER_EXPORTER.getName(), jaegerContainer.getContainerId(),
+                    new ContainerShutdownCloseable(jaegerContainer, "Jaeger"),
+                    OTEL_CONFIG_ENDPOINT, jaegerHost);
+        };
+
+        return jaegerContainerLocator
+                .locateContainer(capturedDevServicesConfiguration.serviceName(), capturedDevServicesConfiguration.shared(),
+                        LaunchMode.current())
+                .map(containerAddress -> {
+                    String jaegerUrl = JAEGER_SCHEME + containerAddress.getUrl();
+                    return new RunningDevService(Feature.OPENTELEMETRY_JAEGER_EXPORTER.getName(), containerAddress.getId(),
+                            null, OTEL_CONFIG_ENDPOINT, jaegerUrl);
+                })
+                .orElseGet(defaultJaegerSupplier);
+    }
+
+    private static class QuarkusJaegerContainer extends GenericContainer<QuarkusJaegerContainer> {
+        private final Integer consolePort;
+
+        private final OptionalInt fixedExposedPort;
+        private final boolean useSharedNetwork;
+
+        private String hostName = null;
+
+        public QuarkusJaegerContainer(DockerImageName dockerImageName, Integer consolePort, OptionalInt fixedExposedPort,
+                String serviceName, boolean useSharedNetwork) {
+            super(dockerImageName);
+            this.consolePort = consolePort;
+            this.fixedExposedPort = fixedExposedPort;
+            this.useSharedNetwork = useSharedNetwork;
+
+            this.waitStrategy = (new LogMessageWaitStrategy()).withRegEx(
+                    ".*\"Health Check state change\",\"status\":\"ready\".*")
+                    .withStartupTimeout(Duration.of(15L, ChronoUnit.SECONDS));
+
+            if (serviceName != null) {
+                withLabel(DEV_SERVICE_LABEL, serviceName);
+            }
+        }
+
+        @Override
+        public void close() {
+
```

Comment:
You added `ContainerShutdownCloseable` which doesn't close container, why to call it at the first place? And container keeps running unless it is shut down by RYUK? You don't need to add closeable if you don't want to (use null), but if you terminate process normally in DEV mode, this close would be called and container stopped, that's better than rely on RYUK and it's safety net (f.e. in past folk with Podman used to disable RYUK, now there is better workaround, but still, I don't get it).

Reply:
Sorry, I probably didn't realize my IDE autogenerated methods for overriding, as also `containerIsStarted` method is empty.